core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -609,34 +683,31 @@ public void cancelled(Context context) {
       ServerMethodDefinition<ReqT, RespT> interceptedDef = methodDef.withServerCallHandler(handler);
       ServerMethodDefinition<?, ?> wMethodDef = binlog == null
           ? interceptedDef : binlog.wrapMethodDefinition(interceptedDef);
-      return startWrappedCall(fullMethodName, wMethodDef, stream, headers, context, tag);
+      return wMethodDef;
+    }
+
+    private final class ServerCallParameters<ReqT, RespT> {
+      ServerCallImpl<ReqT, RespT> call;
+      ServerMethodDefinition<ReqT, RespT> methodDef;
+
+      public ServerCallParameters(ServerCallImpl<ReqT, RespT> call,
+                                  ServerMethodDefinition<ReqT, RespT> methodDef) {
+        this.call = call;
+        this.methodDef = methodDef;
+      }
     }
 
     private <WReqT, WRespT> ServerStreamListener startWrappedCall(
         String fullMethodName,
```

Comment:
If you are interested: Looks like we can remove this argument by using `params.call.getMethodDescriptor().getFullMethodName()`. Looks like ServerCallHandled then wouldn't need to look at the fullMethodName at all (only MethodLookup would).

Reply:
oh looks this is only for logging the original fullMethodName. And this is an existing code path.